sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Storage.Files.Shares.Models
+{
+    /// <summary>
+    /// Stats for the share.
+    /// </summary>
+    public partial class ShareStatistics
+    {
+
+
+        /// <summary>
+        /// Warning: Share usage may exceed int.MaxValue.  Use ShareUsageInBytes instead.
+        /// </summary>
+        public int ShareUsageBytes {
+            get
+            {
+                if (ShareUsageInBytes > int.MaxValue)
+                {
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
+                    throw new OverflowException("ShareUsageInBytes exceeds int.MaxValue.  Use ShareUsageInBytes instead.");
+#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
+                }
+                return (int)ShareUsageInBytes;
+            }
+            internal set { ShareUsageInBytes = value; }
+        }
+    }
+
+    /// <summary>
+    /// ShareModelFactory provides utilities for mocking.
+    /// </summary>
+    public static partial class ShareModelFactory
+    {
+        /// <summary>
+        /// Creates a new PermissionInfo instance for mocking.
+        /// </summary>
+        public static ShareStatistics ShareStatistics(int shareUsageBytes)
```

Comment:
Nit - use [EditorBrowsable(Never)] here so folks don't even see this in intellisense.

Reply:
fixed.